jwst/pipeline/calwebb_coron3.py

```diff
@@ -76,44 +74,37 @@ def process(self, input):
             psf_models.append(psf_input)
             psf_input.close()
 
-        # Call the stack_refs step to stack all the PSF images into
-        # a single CubeModel
+        # Stack all the PSF images into a single CubeModel
         psf_stack = self.stack_refs(psf_models)
         psf_models.close()
 
         # Save the resulting PSF stack
-        output_file = mk_prodname(self.output_dir, prod['name'], 'psfstack')
-        log.info('Saving psfstack file %s', output_file)
-        psf_stack.save(output_file)
+        psf_stack.meta.filename = prod['name']
+        self.save_model(psf_stack, suffix='psfstack')
 
         # Call the sequence of steps align_refs, klip, and outlier_detection
         # once for each input target exposure
         resample_input = datamodels.ModelContainer()
         for target_file in targ_files:
 
             # Call align_refs
-            log.debug('Calling align_refs for member %s', target_file)
+            self.log.debug('Calling align_refs for member %s', target_file)
             psf_aligned = self.align_refs(target_file, psf_stack)
 
             # Save the alignment results
-            filename = mk_filename(self.output_dir, target_file, 'psfalign')
-            log.info('Saving psfalign file %s', filename)
-            psf_aligned.save(filename)
+            self.save_model(psf_aligned, suffix='psfalign')
 
             # Call KLIP
-            log.debug('Calling klip for member %s', target_file)
-            #psf_sub, psf_fit = self.klip(target_file, psf_aligned)
+            self.log.debug('Calling klip for member %s', target_file)
             psf_sub = self.klip(target_file, psf_aligned)
             psf_aligned.close()
 
             # Save the psf subtraction results
-            filename = mk_filename(self.output_dir, target_file, 'psfsub')
-            log.info('Saving psfsub file %s', filename)
-            psf_sub.save(filename)
+            self.save_model(psf_sub, suffix='psfsub')
```

Comment:
Does the psf_sub data model have the proper filename value in its meta data, equivalent to 'target_file'?

Reply:
Yes, it gets it from `self.output_basename`.

Reply:
Correction.  `meta.filename` is populated with the filename of the input file in the association, but of course if you supply the suffix to `save_model`, the old suffix is automatically replaced with the new one.